### PR TITLE
chore!: remove old template endpoints [DET-9337]

### PR DESCRIPTION
**Pull request by hamidzr**
## Test Plan

create a template, list with and without details, describe it, remove it

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/release-notes/remove-old-tempalte-apis.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**Breaking Change**
+
+-  Old template CRUD endpoints have been removed. These used to be serviced under `/templates/*`.
+   Please use the new apis located at `/api/v1/templates/*` instead.
```
Comment (tara-det-ai) [inline docs/release-notes/remove-old-tempalte-apis.rst:6]: The previous template CRUD endpoints have been removed from the /templates/* location.
The new APIs can be found at /api/v1/templates/*.

**Comment by tara-det-ai** [comment]
one suggested edit https://github.com/determined-ai/determined/pull/6572/files#r1170060174

File: docs/release-notes/remove-old-tempalte-apis.rst (hunk 1 above)
Comment (hamidzr) [reply docs/release-notes/remove-old-tempalte-apis.rst:6]: thanks, that sounds good. The second sentence could be misinterpreted I think. maybe we should remove "new" from that? 
I want to make sure the reader won't think that the new apis are added here and that they did exist before and people should use those if it affetcs them

**Review by maxrussell — APPROVED** [review]

**Comment by hamidzr** [comment]
I'm not sure what conventional commit prefix to use here

**Pull request merged**